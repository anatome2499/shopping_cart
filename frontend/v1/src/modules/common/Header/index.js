import Header from './Header'
import { withStyles } from '@material-ui/core'
import styles from './styles'


export default withStyles(styles)(Header)