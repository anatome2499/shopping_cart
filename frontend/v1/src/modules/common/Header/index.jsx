import React, { useState } from 'react'
import {withStyles, Paper, Grid} from '@material-ui/core'
import styles from './styles'
import ButtonWithIcon from '../ButtonWithIcon'
import SelectWithDropdown from '../SelectWithDropdown'
import SearchInput from '../SearchInput'

const Header = ({classes, amount, handleSubmit}) => {
  const [searchValue, setSearchValue] = useState('')
  const [filter, setFilter] = useState('')
  const handleSearchChange = event => {
    setSearchValue(event.target.value)
  }
  const [age, setAge] = React.useState('')
  const handleAgeChange = (event) => {
    setAge(event.target.value);
  }
  const handleSearch = () => {
    handleSubmit(searchValue, filter, age)
  }


  return (
    <Paper className={classes.root}>
      <Grid container spacing={1}>
        <Grid item xs={10}>
          <Grid container spacing={1}>
            <Grid item xs={12}>
              <SelectWithDropdown handleChange={handleAgeChange} value={age} />
            </Grid>
            <Grid item xs={12}>
              <SearchInput id={'search-field'} value={searchValue} handleChange={handleSearchChange} handleButtonClick={handleSearch} />
            </Grid>
          </Grid>
        </Grid>
        <Grid item xs={2} style={{justifyContent: 'flex-end'}} >
          <ButtonWithIcon amount={amount ? amount : 0} />
        </Grid>
      </Grid>
    </Paper>
  )
}

export default withStyles(styles)(Header)