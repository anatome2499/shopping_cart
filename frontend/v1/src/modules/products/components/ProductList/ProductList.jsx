import React from 'react'
import Grid from '@material-ui/core/Grid'

import ProductCard from '../ProductCard'

const ProductList = ({
  productItems = []
}) => {

  return (
    <Grid container spacing={5}>
      {productItems.map(item => (
        <Grid
          item
          xs={12}
          sm={4}
          key={item.id}
        >
          <ProductCard
            
            {...item}
          />
        </Grid>
      ))}
    </Grid>
  )
}

export default ProductList