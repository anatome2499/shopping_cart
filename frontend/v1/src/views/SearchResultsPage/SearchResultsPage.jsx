import React from 'react'

import Header from '../../modules/common/Header'
import ProductList from '../../modules/products/components/ProductList'

const SearchResultsPage = ({productItems = [], filter}) => {
    const handleSubmit = (searchValue, filter, age) => {
        // TODO WAIT API
    }

    return (
        <div>
            <Header amount={0} handleSubmit={handleSubmit} />
            <ProductList productItems={productItems} />
        </div>
    )
}

export default SearchResultsPage
