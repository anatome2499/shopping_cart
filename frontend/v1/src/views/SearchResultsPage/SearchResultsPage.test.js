import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import SearchResultsPage from './SearchResultsPage'
import Header from '../../modules/common/Header'
import ProductList from '../../modules/products/components/ProductList'


describe('<SearchResultsPage />', () => {
  let component

  beforeEach(() => {
    component = shallow(<SearchResultsPage />)

  })

  it('should render with no errors', () => {
    
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should render with Header', () => {
    expect(toJson(component)).toMatchSnapshot()

    expect(component.find(Header).length).toEqual(1)
  })

  it('should render with ProductList', () => {
    expect(toJson(component)).toMatchSnapshot()

    expect(component.find(ProductList).length).toEqual(1)
  })
})