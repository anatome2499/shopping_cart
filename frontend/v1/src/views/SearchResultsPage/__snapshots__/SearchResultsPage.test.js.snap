// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResultsPage /> should render with Header 1`] = `
<div>
  <WithStyles(Header) />
  <ProductList />
</div>
`;

exports[`<SearchResultsPage /> should render with ProductList 1`] = `
<div>
  <WithStyles(Header) />
  <ProductList />
</div>
`;

exports[`<SearchResultsPage /> should render with no errors 1`] = `
<div>
  <WithStyles(Header) />
  <ProductList />
</div>
`;
